app/Http/Controllers/CollectionController.php

```diff
@@ -121,6 +121,15 @@ public function viewPublicJson(Request $request, $collectionID)
                 ];
             }
         }
+
+        if(count($data['datasets']) == 0){
+            $collection->description = null; 
+            $collection->warning .= "<p>0 results found</p>";
+            $data['metadata']['warning'] =  $collection->warning ;

```

Comment:
Please concatenate the warning in the `data['metadata']['warning']` instead of the model property. In case in the future development accidentally save the model.